src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java

```diff
@@ -188,6 +194,53 @@ public void handle(Endpoint<?> endpoint) {
                     sendConsumerCommitOffsetResponse(routingContext.response(), status.succeeded());
                 });
                 break;
+
+            case SEEK:
+
+                JsonArray seekOffsetsList = bodyAsJson.getJsonArray("offsets");
+
+                List<Future> seekHandlers = new ArrayList<>(seekOffsetsList.size());
+                for (int i = 0; i < seekOffsetsList.size(); i++) {
+                    TopicPartition topicPartition = new TopicPartition(seekOffsetsList.getJsonObject(i));
+                    long offset = seekOffsetsList.getJsonObject(i).getLong("offset");
+                    Future<Void> fut = Future.future();
+                    seekHandlers.add(fut);
+                    this.seek(topicPartition, offset, fut.completer());
+                }
+
+                CompositeFuture.join(seekHandlers).setHandler(done -> {
+                    if (done.succeeded()) {
+                        sendSeekResponse(routingContext.response(), ErrorCodeEnum.NO_CONTENT);
+                    } else {
+                        sendSeekResponse(routingContext.response(), ErrorCodeEnum.INTERNAL_SERVER_ERROR);
+                    }
+                });
+                break;
+
+            case SEEK_TO_BEGINNING:
+            case SEEK_TO_END:
+
+                JsonArray seekPartitionsList = bodyAsJson.getJsonArray("partitions");
+
+                Set<TopicPartition> set = seekPartitionsList.stream()
+                        .map(JsonObject.class::cast)
+                        .map(json -> new TopicPartition(json.getString("topic"), json.getInteger("partition")))
+                        .collect(Collectors.toSet());
+
+                Handler<AsyncResult<Void>> seekHandler = done -> {
```

Comment:
This is the same handler as for `SEEK` above, so why not factor out a common class for it?

Reply:
common class or common method?

Reply:
A common method. (It will result in a common anonymous class anyway)

Reply:
Why an anonymous class, I don't see what you mean. I can extract a method with `done` as a param.

Reply:
I have just factored out seek(s) methods for now. Related to the comment on seek handler, I have an issue about refactoring around sending responses to the client. I will revisit this point while addressing that issue (#180)